### ISSUE TITLE
coil-avif multiplatform rendering

### ISSUE THREAD

**Issue by vanniktech** [coil-kt/coil#2742]
I saw that AVIF support on Android 12+ was added here: https://github.com/coil-kt/coil/pull/1669

Would it be possible to also have this working on previous Android versions? Additionally with also support for all the other platforms (ios/wasm/desktop)? This could live in a new module like `coil-svg`

AVIF offers superior compression compared to JPEG and WebP, both in lossy and lossless modes, with support for transparency and high dynamic range (HDR). The only problem is that it isn't widely supported yet.

**Comment by colinrtwhite** [comment]
That PR doesn't add support for AVIF - it just ensures we can decode an AVIF image using Coil on Android 12+ using the standard Android decoding classes (`BitmapFactory`/`ImageDecoder`). The decoders are built into Android 12+ so we can't backport them.

To support pre-Android 12 and multiplatform we'd have to import and wrap a 3rd party AVIF decoder, which would require importing native libraries. It looks like Glide has [an integration for Android](https://github.com/bumptech/glide/blob/master/integration/avif/src/main/java/com/bumptech/glide/integration/avif/AvifByteBufferBitmapDecoder.java) that already does this, but it would need more work to integrate with all multiplatform targets I believe. It should be possible to write a custom Coil `Decoder` that wraps the same library like Glide does and add it to your `ComponentRegistry`.

If there's enough demand we could consider adding a 1st party Coil library to support this, but we'd need to figure out how to support this on multiplatform first. I'm hoping Skia adds support for AVIF as it would simplify the integration, but I'm not sure if there are plans to do that.

**Comment by colinrtwhite** [comment]
Someone backported AVIF support to API 24 on Android! https://github.com/awxkee/avif-coder-coil

**Comment by awxkee** [comment]
> It looks like Glide has [an integration for Android](https://github.com/bumptech/glide/blob/master/integration/avif/src/main/java/com/bumptech/glide/integration/avif/AvifByteBufferBitmapDecoder.java) that already does this,

Glide integration do not support HDR images and ICC profiles.

TBH basic AVIF support is relatively easy to implement. However, support of all features especially those it was designed for (HDR, color profiles, upcoming Gain maps), requires tremendous amount of work, particularly when ensuring compatibility across multiple platforms.

> I'm hoping Skia adds support for AVIF as it would simplify the integration, but I'm not sure if there are plans to do that.

Skia definetly [supports AVIF](https://github.com/google/skia/blob/main/src/codec/SkAvifCodec.cpp). However, if you're using a wrapper for Skia, it may not have been built with AVIF support enabled. I hadn't look deeply at it, however from skimming looks like Skia also misses HDR support.

**Comment by xiaoyvyv** [comment]
Here is a multi-platform library for reference: https://github.com/qdsfdhvh/avif-kmp

**Comment by YektaDev** [comment]
Since it’s been some time, I thought I’d ask if there’s any plans/updates on this.

**Comment by hellosagar** [comment]
Yep still waiting if there is an AVIF support coming to coil for Android and iOS

**Comment by anggrayudi** [comment]
There's already a library that can render AVIF on older versions of iOS and Android:

https://github.com/alfikri-rizky/AvifKit

Maybe we can try this
